proposals/simd-0335-parent-ready-update-marker.md

```diff
@@ -0,0 +1,229 @@
+---
+simd: '0337'
+title: ParentReadyUpdate Block Marker
+authors:
+  - ksn6 (Anza)
+category: Standard
+type: Core
+status: Review
+created: 2025-08-13
+feature: TBD
+---
+
+## Summary
+
+This SIMD proposes upgrading `BlockMarkerV1`, introduced in SIMD-0307, to
+`BlockMarkerV2` to support optimistic block packing in the Alpenglow consensus
+protocol. In particular, `BlockMarkerV2` includes a new `ParentReadyUpdate`
+variant to support optimistic block packing. This marker signals when a leader
+has switched to a different parent block during optimistic block production
+due to a `ParentReady` event, providing critical metadata for consensus
+validation and block verification.
+
+## Motivation
+
+The Alpenglow consensus protocol, as specified in SIMD-0326, introduces
+optimistic block packing as a key performance optimization. This mechanism
+allows leaders to begin constructing blocks on tentative parent blocks before
+the actual parent is fully finalized through the notarization process. This
+optimization significantly reduces block production latency by enabling
+parallel processing of block construction and parent finalization.
+
+However, during optimistic block packing, a leader may need to switch from
+their initially assumed parent to a different parent block when a
+`ParentReady` event occurs. This event indicates that:
+
+1. All necessary skip certificates have been collected for intervening slots
+2. A `NotarizeFallback` certificate exists for a valid parent block
+3. The leader must switch to this newly ready parent (as specified in
+   Algorithm 3, Line 7 of the
+   [Alpenglow whitepaper](https://www.anza.xyz/alpenglow-1-1))
+
+
+Currently, there is no standardized mechanism to signal these parent switches
+within the block structure. As such, without the capability that this SIMD
+details, leader validators cannot indicate that a parent switch has occurred
+during optimistic packing.
+
+This SIMD addresses these issues by leveraging the extensible `BlockMarker`
+framework introduced in SIMD-0307 to add a dedicated marker for parent switch
+events.
+
+## New Terminology
+
+- **Block Marker**: A chunk of structured non-transaction data that can be
+  placed before, after, or in-between entry batches in a block.
+- **Optimistic Block Packing**: A technique where leaders begin constructing
+  blocks on a tentative parent before the actual parent is finalized
+- **ParentReady Event**: A consensus event indicating that a block has met
+  all requirements to serve as a parent for new block production
+- **Parent Switch**: The action of changing from one parent block to another
+  during optimistic block production
+- **ParentReadyUpdate**: A block marker that records when and to which parent
+  a switch occurred during optimistic packing
+
+## Detailed Design
```

Comment:
might have missed it but good to call out the failure cases:
- multiple ParentReadyUpdate's within the same block
- erroneous values (e.g. parent slot is >= current slot)
and what we do in such cases - do not attempt to continue replay, mark the block as dead